### DISCUSSION TITLE
Error in Random Forest

### DISCUSSION THREAD

**Discussion by yohanespm77** [teamalgoritma/community#398] [Q&A] [answered]
Error ketika ingin melakukan prediksi terhadap data.

Kode saya : 
data_pred <- read.delim("clipboard", stringsAsFactors = T)
data_pred[,c("Source.IP", "Destination.IP", "Signature.IDS..custom.")] <- lapply(data_pred[, c("Source.IP", "Destination.IP", "Signature.IDS..custom.")], as.factor)
pred <- predict(modelRF, data_pred)

Errornya :
Error in predict.randomForest(modelRF, data_pred) : 
  New factor levels not present in the training data

Data training dan data yg akan diprediksi sudah sesuai tipe datanya dan variabel-variabelnya.

**Comment by jafarhf1** [comment, marked as answer]
Dear @yohanespm77, 
 
Data training bapak ada yang berformat `factor`, dimana `factor` itu sebenarnya kategori. Misalkan data training bapak hanya memiliki 3 kategori, pada `data_pred` terdapat 4 kategori. Ada kategori yang belum pernah muncul, dan itu menyebabkan model machine learning tidak bisa melakukan prediksi. 

Hal yang bapak harus lakukan adalah memastikan semua levels pada data berformat factor sudah sama semua.
